src/main/java/Storage.java

```diff
@@ -0,0 +1,107 @@
+import java.io.*;
+
+public class Storage {
+    private final String pathName;
+    private final String fileName;
+
+    public Storage(String pathName, String fileName) {
+        this.pathName = pathName;
+        this.fileName = fileName;
+    }
+
+    public File initialiseDirectory() throws DukeException {
+        File directory = new File(pathName);
+        boolean hasDirectory = directory.exists();
+
+        if (!hasDirectory) {
+            hasDirectory = directory.mkdir();
+        }
+
+        if (hasDirectory) {
+            return directory;
+        } else {
+            throw new DukeException("\t" + "Unable to initialise directory");
+        }
+    }
+
+    public File initialiseFile(File directory) throws IOException {
+        File file = new File(directory + "/" + fileName);
+        boolean hasFile = file.exists();
+
+        if (!hasFile) {
+            hasFile = file.createNewFile();
+        }
+
+        if (hasFile) {
+            return file;
+        } else {
+            throw new IOException("\t" + "Unable to initialise file");
+        }
+    }
+
+    public void loadTasksFromFile(File dataFile, TaskList tasks) throws IOException {
+        FileReader fileReader = new FileReader(dataFile);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        String line;
+
+        while ((line = bufferedReader.readLine()) != null) {
+            String[] task = line.trim().split("\\|");
```

Comment:
Plural should be used in this case, perhaps consider changing to `tasks`?
I noticed the same issue in several other places too, I will not repeat this further.